lib/monitoring/aws-lambda/LambdaFunctionMonitoring.ts

```diff
@@ -493,6 +506,12 @@ export class LambdaFunctionMonitoring extends Monitoring {
       this.addAlarm(createdAlarm);
     }
     for (const disambiguator in props.addMaxIteratorAgeAlarm) {
+      if (!this.isIterator) {
+        throw new Error(
+          "addMaxIteratorAgeAlarm is not applicable if isIterator is not true",
+        );
```

Comment:
I feel I've asked this before, so sorry in advance if i have

Can we use `Annotations` here? Having the scope included (via `Annotations.error(...)`) really helps folks find where in their app this is being thrown